x-pack/plugin/sql/jdbc/src/main/java/org/elasticsearch/xpack/sql/jdbc/JdbcDatabaseMetaData.java

```diff
@@ -1245,37 +1245,31 @@ static String escapeStringPattern(String pattern, EscapeWildcard strategy) {
             default:
                 // first detect if there's any escaping in the string, if so bail out
                 boolean escaped = false;
-                boolean hasEscaping = false;
 
                 for (int i = 0; i < pattern.length(); i++) {
                     char curr = pattern.charAt(i);
-                    if (escaped == false && (curr == esc) && esc != 0) {
+                    if (escaped == false && curr == esc) {
                         escaped = true;
                     } else {
                         if ((curr == '%' || curr == '_' || curr == esc) && (escaped == true)) {
-                            hasEscaping = true;
-                            break;
+                            // looks like the string is using escaping, bail out
+                            return pattern;
                         }
                         escaped = false;
                     }
                 }
 
-                // looks like the string is using escaping, bail out
-                if (hasEscaping) {
-                    return pattern;
-                }
-
+                escaped = false;
```

Comment:
Can we have a test case for this?

Reply:
Added a few in the `JdbcDatabaseMetaDataTests`

Reply:
thank you!